src/Stryker.Core/Stryker.Core/MutationTest/MutationTestProcess.cs

```diff
@@ -186,7 +188,27 @@ public StrykerRunResult Test(StrykerOptions options)
                 }
             }
 
-            var mutantsToTest = mutantsNotRun.Where(x => x.ResultStatus != MutantStatus.Ignored && x.ResultStatus != MutantStatus.NoCoverage);
+            var mutantsToTest = mutantsNotRun;
+            if (_options.Optimizations.HasFlag(OptimizationFlags.CoverageBasedTest))
+            {
+                var testCount = _mutationTestExecutor.TestRunner.DiscoverNumberOfTests();
+                var toTest = @mutantsNotRun.Sum(x => x.MustRunAgainstAllTests ? testCount : x.CoveringTests.Count);
+                var total = testCount * viableMutantsCount;
+                if (total > 0 && total != toTest)
+                {
+                    _logger.LogInformation($"Coverage analysis eliminated {(total-toTest)/(double)total:P1} useless tests.");
+                }
+            }
+            else if (_options.Optimizations.HasFlag(OptimizationFlags.SkipUncoveredMutants))
+            {
+                var total = viableMutantsCount;
+                var toTest = mutantsToTest.Count();
+                if (total > 0  && total!= toTest)
+                {
+                    _logger.LogInformation($"Coverage analysis eliminated {(total-toTest)/(double)total:P1} useless tests.");
```

Comment:
```suggestion
                    _logger.LogInformation($"Coverage analysis eliminated {(total-toTest)/(double)total:P1} of tests because they do not cover a mutant.");
```

Reply:
Here, the strategy is simply to removed non covered mutants so we can skip testing the associated tests. so the proposed message is not a true descritption

Reply:
I'm confused, if a mutant is not covered, how can it have associated tests?